apis/controlplane/v1alpha4/nestedcontrolplane_types.go

```diff
@@ -28,6 +29,9 @@ type NestedControlPlaneSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
+	// EtcdRef is the reference to the NestedEtcd
+	EtcdRef *corev1.ObjectReference `json:"etcd,omitempty"`
+
```

Comment:
Opps, we'll have a couple conflicts between https://github.com/kubernetes-sigs/cluster-api-provider-nested/pull/38 and this but we can update them accordingly.

Reply:
Will modify this field based on #38